gitpod-io/gitpod
# [inner loop poc] debug workspace window

## Description
This PR introduces an **experimental approach** to debug gitpod workspace configuration. To be tested internally first, this will allow testing `.gitpod.yml` and `.gitpod.Dockerfile` within a workspace without the need to `git push` such files.

For demo, see [an internal thread](https://gitpod.slack.com/archives/C01KGM9DVRC/p1671818659031569).

The debug location:
- does not use blobserve to serve static content, but serve directly from the workspace
- does not have access to sensitive cookies like Gitpod cookie

This PR as well fixes service VS Code production built from sources (not blobserve) and align serving of webviews for production (workspace locations) and in dev mode (port location).

## Related Issue(s)

Fixes #

## How to test


**Test for regressions**

- Start VS Code stable, check that everything works and the inner loop actions are NOT available.

**Test for feature flag**

- Start VS Code latest, check that everything works and the inner loop actions are NOT available.
- Enable the feature flag for a user.
- Start VS Code latest, check that everything works and the inner loop actions are available.

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
